operations/iuf/workflows/upgrade_all_products.md

```diff
@@ -392,20 +392,250 @@ Once this step has completed:
 
 This section describes how to update software on management nodes. It describes how to test a new image and CFS configuration on a single "canary node" first before rolling it out to the other management nodes. Modify the procedure
 as necessary to accommodate site preferences for rebuilding management nodes. The images and CFS configurations used are created by the `prepare-images` and `update-cfs-config` stages respectively; see the `prepare-images`
-[Artifacts created](../stages/prepare_images.md#artifacts-created) section for details on how to query the images and CFS configurations and [update-cfs-config](../stages/update_cfs_config.md) section for details about how the CFS configuration is updated.
+[Artifacts created](../stages/prepare_images.md#artifacts-created) section for details on how to query the images and [update-cfs-config](../stages/update_cfs_config.md) section for details about how the CFS configuration is updated.
 
 **`NOTE`** Additional arguments are available to control the behavior of the `management-nodes-rollout` stage, for example `--limit-management-rollout` and `-cmrp`. See the
 [`management-nodes-rollout` stage documentation](../stages/management_nodes_rollout.md) for details and adjust the examples below if necessary.
 
-**`NOTE`** The `management-nodes-rollout` stage creates additional separate Argo workflows when rebuilding NCN worker nodes. The Argo workflow names will include the string `ncn-lifecycle-rebuild`. If monitoring progress with the Argo UI,
-remember to include these workflows.
+**Important** There is a different procedure for  `management-nodes-rollout` depending on whether or not CSM is being upgraded.
+The two procedures differ in the handling of NCN storage nodes and NCN master nodes. If CSM is not being upgraded, then NCN storage nodes and NCN master nodes will not be upgraded and will be updated by the CFS configuration
+created in [update-cfs-config](../stages/update_cfs_config.md).
+If CSM is being upgraded, the NCN storage nodes and NCN master nodes will be upgraded.
+Both procedures use the same steps for rebuilding/upgrading NCN worker nodes.
+Select **one** of the following procedures based on whether or not CSM is being upgraded.
+
+- [Management-nodes-rollout with CSM upgrade](#651-management-nodes-rollout-with-csm-upgrade)
```

Comment:
I think this section (and the one referenced in the line right below this one) is really about the master and storage nodes only, right?  If so, I'd reword as
```
NCN master and storage nodes with CSM upgrade scenario
```

That way, the flow is a bit more obvious/consistent, e.g the table of contents at the top will look like this:
```
6.5 Execute the IUF `management-nodes-rollout` stage
- 6.5.1 NCN master and strorage nodes with CSM upgrade
- 6.5.2 NCN master and storage nodes without CSM upgrade
- 6.5.3 NCN worker nodes
```

Or maybe it should even be:
```
6.5 Execute the IUF `management-nodes-rollout` stage
- 6.5.1 NCN master and strorage nodes
    - 6.5.1.1 with CSM upgrade
    - 6.5.1.2 without CSM upgrade
- 6.5.2 NCN worker nodes
```